src/domain/opendatafabric/src/identity/grammar.rs

```diff
@@ -172,24 +172,23 @@ impl Grammar {
 
     // DatasetAliasRemote = RepoName "/" (AccountName "/")? DatasetName
     pub fn match_dataset_alias_remote(s: &str) -> Option<(&str, Option<&str>, &str, &str)> {
-        match s.split_once('/') {
-            None => None,
-            Some((repo, rest)) => match Self::match_repo_name(repo) {
-                Some((repo, "")) => match rest.split_once('/') {
-                    Some((acc, rest)) => match Self::match_account_name(acc) {
-                        Some((acc, "")) => {
-                            let (ds, tail) = Self::match_dataset_name(rest)?;
-                            Some((repo, Some(acc), ds, tail))
-                        }
-                        _ => None,
-                    },
-                    None => {
-                        let (ds, tail) = Self::match_dataset_name(rest)?;
-                        Some((repo, None, ds, tail))
-                    }
-                },
-                _ => None,
-            },
-        }
+        let (repo, rest) = s.split_once('/')?;
+
+        let Some((repo, "")) = Self::match_repo_name(repo) else {
+            return None;
+        };
+
+        let Some((acc, rest)) = rest.split_once('/') else {
+            let (ds, tail) = Self::match_dataset_name(rest)?;
+            return Some((repo, None, ds, tail));
+        };
+
+        let Some((acc, "")) = Self::match_account_name(acc) else {
+            return None;
+        };
+
+        let (ds, tail) = Self::match_dataset_name(rest)?;
+
+        Some((repo, Some(acc), ds, tail))
```

Comment:
I don't see any positive gain from this rule. Previous structure  was easier to follow

Reply:
The rule was disabled; these changes were reverted